admin-ui/plugins/auth-server/components/Agama/AgamaListPage.js

```diff
@@ -0,0 +1,362 @@
+import React, { useState, useEffect, useContext, useCallback } from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+import { Card, CardBody, Badge, Input } from 'Components'
+import applicationStyle from 'Routes/Apps/Gluu/styles/applicationstyle'
+import { useTranslation } from 'react-i18next'
+import SetTitle from 'Utils/SetTitle'
+import { ThemeContext } from 'Context/theme/themeContext'
+import getThemeColor from 'Context/theme/config'
+import { Paper, TablePagination } from '@material-ui/core'
+import { getAgama, deleteAgama } from '../../redux/actions/AgamaActions'
+import { hasPermission, AGAMA_READ, AGAMA_WRITE } from 'Utils/PermChecker'
+import axios from 'axios'
+import GluuViewWrapper from '../../../../app/routes/Apps/Gluu/GluuViewWrapper'
+import MaterialTable from '@material-table/core'
+import { Button, Modal, ModalBody, ModalFooter, ModalHeader } from 'reactstrap'
+import { useDropzone } from 'react-dropzone'
+import JSZip from 'jszip'
+const JansConfigApi = require('jans_config_api')
+import { getClient } from '../../../../app/redux/api/base'
+import { AGAMA_DELETE } from '../../../../app/utils/PermChecker'
+import {CircularProgress} from '@material-ui/core'
+
+function AgamaListPage() {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+  const userAction = {}
+  const options = {}
+  const clientOptions = {}
+  const myActions = []
+  const [limit, setLimit] = useState(10)
+  const [pageNumber, setPageNumber] = useState(0)
+  const [showAddModal, setShowAddModal] = useState(false)
+
+  const [selectedFile, setSelectedFile] = useState(null)
+  const [projectName, setProjectName] = useState('')
+  const [getProjectName, setGetProjectName] = useState(false)
+  const [selectedFileName, setSelectedFileName] = useState(null)
+  const [shaFile, setSHAfile] = useState(null)
+  const [shaStatus, setShaStatus] = useState(false)
+  const [shaFileName, setShaFileName] = useState('')
+  const [localLoading, setLocalLoading] = useState(false)
+
+
+  const token = useSelector((state) => state.authReducer.token.access_token)
+  const issuer = useSelector((state) => state.authReducer.issuer)
+
+  const BASE_PATH = getClient(JansConfigApi, token, issuer)
+
+  const theme = useContext(ThemeContext)
+  const selectedTheme = theme.state.theme
+  const themeColors = getThemeColor(selectedTheme)
+  const bgThemeColor = { background: themeColors.background }
+  function convertFileToByteArray(file) {
+    return new Promise((resolve, reject) => {
+      const reader = new FileReader()
+      reader.readAsArrayBuffer(file)
+      reader.onload = () => {
+        const byteArray = new Uint8Array(reader.result)
+        resolve(byteArray)
+      }
+      reader.onerror = (error) => {
+        reject(error)
+      }
+    })
+  }
+
+  const submitData = async () => {
```

Comment:
We should call any API  using redux-saga (see `app\redux\api\backend-api.js`) . Can we call the post endpoint from AgamaApi.js file using redux saga.